### PR TITLE
Support log forwarding customisation for groups of namespaces

### PR DESCRIPTION
**Pull request by DebakelOrakel**
## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [ ] Link this PR to related issues.

### CODE REVIEW

**Comment by DebakelOrakel** [comment]
would be breaking, as the configuration of existing logforwarders need to be updated.

File: component/main.jsonnet [hunk 1]
```diff
@@ -69,38 +69,43 @@ local alert_rules = import 'alertrules.libsonnet';
...
-            namespaces: [ ns ],
+            namespaces: params.clusterLogForwarding.namespace_groups[group].namespaces,
           },
         }
-        for ns in std.objectFields(params.clusterLogForwarding.namespaces)
+        for group in std.objectFields(params.clusterLogForwarding.namespace_groups)
       ],
-      [if std.length(params.clusterLogForwarding.namespaces) > 0 then 'pipelines']: [
+      [if std.length(params.clusterLogForwarding.namespace_groups) > 0 then 'pipelines']: [
+        local enable_json = com.getValueOrDefault(params.clusterLogForwarding.namespace_groups[group], 'json', false);
```
Comment (simu) [inline component/main.jsonnet:82]: This PR seems to make two logically distinct changes:

1. Refactor the config to expose namespace groups instead of namespaces for log forwarding
1. Add support for enabling JSON log parsing for custom forwarding

To have a nice changelog, it would make sense to separate those two changes into their own PRs. Additionally, having such changes in separate PRs would make reviewing the changes much easier.

File: component/main.jsonnet [hunk 2]
```diff
@@ -69,38 +69,43 @@ local alert_rules = import 'alertrules.libsonnet';
...
-        },
+          name: group,
+          inputRefs: params.clusterLogForwarding.namespace_groups[group].namespaces,
+          outputRefs: com.getValueOrDefault(params.clusterLogForwarding.namespace_groups[group], 'forwarders', []),
+        } + com.makeMergeable(if enable_json then patch_json else {})
+        for group in std.objectFields(params.clusterLogForwarding.namespace_groups)
       ],
-    }),
+    } + com.makeMergeable(
+      local enable_json = com.getValueOrDefault(params.clusterLogForwarding.application_logs, 'json', false);
```
Comment (simu) [inline component/main.jsonnet:92]: This can be made backwards-compatible very easily:

```suggestion
      local enable_json = com.getValueOrDefault(
        params.clusterLogForwarding.application_logs,
        'json',
        params.clusterLogForwarding.json.enabled
      );
```

**Review by simu — CHANGES_REQUESTED** [review]
I think the implementation for the new parameter format is ok (as far as I'm able to tell), but from what I can tell, it should be fairly straight-forward to make the change fully backwards-compatible (see my inline comments) in accordance with our best practices, cf. https://syn.tools/syn/explanations/commodore-components/major.html#_structural_changes_of_a_components_existing_parameters.

Additionally, as noted inline, the PR seems to contain two logically separate changes (parameter restructuring and more fine-grained JSON parsing control). To have a nice changelog, it would make sense to separate those two changes into their own PRs. As a side note, having logical changes in separate PRs would make reviewing much easier especially since the two changes are interwoven quite heavily in the implementation.

File: component/main.jsonnet (hunk 1 above)
Comment (DebakelOrakel) [reply component/main.jsonnet]: true, but these changes are very intertwined. however i removed the json part and do a seperate pr

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 3]
```diff
@@ -76,23 +76,51 @@ Each key in this dictionary holds the parameters for an `.spec.outputs` object.
...
 [horizontal]
 type:: dictionary
 default:: {}
 
-Customization for the logging of a specified namespace.
+Customization for the logging of a specified group of namespaces.
+
+Enable forwarders will send the logs of the specified namespaces to a third-party log aggregator. Depending on the need you may need to deploy a separate log forwarder.
+
+Enable json parsing for a 'namespace_group' only makes sense if the logs are forwarded to the clusters default elasticsearch instance. Therefor 'default' will automatically be added to the forwarders.
```
Comment (simu) [inline docs/modules/ROOT/pages/references/parameters.adoc:89]: This probably doesn't belong in this PR anymore

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 4]
```diff
@@ -76,23 +76,51 @@ Each key in this dictionary holds the parameters for an `.spec.outputs` object.
...
+=== `clusterLogForwarding.namespace_groups`
 
 [horizontal]
 type:: dictionary
 default:: {}
 
-Customization for the logging of a specified namespace.
+Customization for the logging of a specified group of namespaces.
+
+Enable forwarders will send the logs of the specified namespaces to a third-party log aggregator. Depending on the need you may need to deploy a separate log forwarder.
```
Comment (simu) [inline docs/modules/ROOT/pages/references/parameters.adoc:87]:
```suggestion
Enabling forwarders will send the logs of the specified namespaces to a third-party log aggregator.
For some log aggregation systems you may need to deploy a separate log forwarder.
```

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 5]
```diff
@@ -76,23 +76,51 @@ Each key in this dictionary holds the parameters for an `.spec.outputs` object.
...
+
+=== `clusterLogForwarding.application_logs`
+
+[horizontal]
+type:: dictionary
+default:: {}
+
+Customization for the logging of all applications.
+
+Enable forwarders will send the logs of all namespaces to a third-party log aggregator. Depending on the need you may need to deploy a separate log forwarder.
```
Comment (simu) [inline docs/modules/ROOT/pages/references/parameters.adoc:114]:
```suggestion
Enabling forwarders will send the logs of all namespaces to a third-party log aggregator.
For some log aggregation systems you may need to deploy a separate log forwarder.
```

**Review by simu — APPROVED** [review]
LGTM now. Some inline comments for the docs
